Issue with Robolectric

I just tried to integrate this with Robolectric 3.0 and tests are failing after the first test with this exception:
```
java.lang.RuntimeException: org.threeten.bp.zone.ZoneRulesException: Unable to register zone as one already registered with that ID: Europe/Sofia, currently loading from provider: TZDB
	at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:244)
	at org.robolectric.RobolectricTestRunner.runChild(RobolectricTestRunner.java:188)
	at org.robolectric.RobolectricTestRunner.runChild(RobolectricTestRunner.java:54)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.RobolectricTestRunner$1.evaluate(RobolectricTestRunner.java:152)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:86)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:49)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:64)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:360)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.threeten.bp.zone.ZoneRulesException: Unable to register zone as one already registered with that ID: Europe/Sofia, currently loading from provider: TZDB
	at org.threeten.bp.zone.ZoneRulesProvider.registerProvider0(ZoneRulesProvider.java:216)
	at org.threeten.bp.zone.ZoneRulesProvider.registerProvider(ZoneRulesProvider.java:201)
	at com.jakewharton.threetenabp.AndroidThreeTen.init(AndroidThreeTen.java:27)
	at com.example.MyApp(MyApp.java:19)
	at org.robolectric.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:140)
	at org.robolectric.RobolectricTestRunner.setUpApplicationState(RobolectricTestRunner.java:433)
	at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:240)
	... 35 more
```

I think the `ZoneRulesProvider` configuration is statically kept from one test to the other. Any idea if there's a way to fix that at ThreetenABP level or is it a robolectric issue?

--- a/build.gradle
+++ b/build.gradle
@@ -22,4 +22,5 @@ ext {
   truth = 'com.google.truth:truth:0.27'
   supportTestRunner = 'com.android.support.test:runner:0.3'
   supportTestRules = 'com.android.support.test:rules:0.3'
+  robolectric = 'org.robolectric:robolectric:3.0'
 }

--- a/sample/build.gradle
+++ b/sample/build.gradle
@@ -6,6 +6,9 @@ dependencies {
   androidTestCompile rootProject.ext.junit
   androidTestCompile rootProject.ext.supportTestRunner
   androidTestCompile rootProject.ext.supportTestRules
+
+  testCompile rootProject.ext.junit
+  testCompile rootProject.ext.robolectric
 }
 
 android {

--- a/sample/src/test/java/com/jakewharton/threetenabp/sample/ExamplesTest.java
+++ b/sample/src/test/java/com/jakewharton/threetenabp/sample/ExamplesTest.java
@@ -1,0 +1,27 @@
+package com.jakewharton.threetenabp.sample;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.annotation.Config;
+import org.threeten.bp.Instant;
+
+import static org.junit.Assert.assertNotNull;
+
+/** This class has two tests to ensure that we can initialize the library multiple times. */
+@Config(constants = BuildConfig.class, sdk = 21)
+@RunWith(RobolectricGradleTestRunner.class)
+public final class ExamplesTest {
+  @Test public void one() {
+    Examples activity = Robolectric.setupActivity(Examples.class);
+    Instant now = activity.now();
+    assertNotNull(now);
+  }
+
+  @Test public void two() {
+    Examples activity = Robolectric.setupActivity(Examples.class);
+    Instant now = activity.now();
+    assertNotNull(now);
+  }
+}

--- a/threetenabp/src/main/java/com/jakewharton/threetenabp/AndroidThreeTen.java
+++ b/threetenabp/src/main/java/com/jakewharton/threetenabp/AndroidThreeTen.java
@@ -12,7 +12,7 @@ public final class AndroidThreeTen {
   private static final AtomicBoolean initialized = new AtomicBoolean();
 
   public static void init(Application application) {
-    if (initialized.compareAndSet(true, true)) {
+    if (initialized.getAndSet(true)) {
       return;
     }
 

